src/realm/deppart/sparsity_impl.cc

```diff
@@ -907,6 +902,19 @@ namespace Realm {
     , sparsity_comm(_sparsity_comm)
   {}
 
+template<int N, typename T>
+SparsityMapImpl<N, T>::~SparsityMapImpl(void)
+{
+     //We are responsible for our instances
+     if (this->from_gpu) {
+       if (get_runtime()->get_memory_impl(this->entries_instance))
+         this->entries_instance.destroy();
+       if (get_runtime()->get_memory_impl(this->approx_instance))
+         this->approx_instance.destroy();
+       this->from_gpu = false;
```

Comment:
Why do you need to set `from_gpu` to be false here? The object is being destructed. Also, Why do you need a `from_gpu` bit in the first place? Isn't that apparent from the existence of the isntances?